aten/src/ATen/native/quantized/cpu/qpool.cpp

```diff
@@ -244,12 +247,142 @@ Tensor quantized_max_pool2d(
 namespace {
 class QMaxPool2D_arr_args final : public torch::OperatorKernel {
  public:
+  #ifdef USE_PYTORCH_QNNPACK
+   Tensor qnnpack_maxpool(
+       Tensor input,
+       IntArrayRef kernel_size,
+       IntArrayRef stride,
+       IntArrayRef padding,
+       IntArrayRef dilation) {
+     Tensor qy;
+
+     TORCH_CHECK(
+         input.ndimension() == 4,
+         "qnnpack_maxpool(): Expected input to be 4-dimensional: got ",
+         input.ndimension());
+     TORCH_CHECK(
+         kernel_size.size() == 2,
+         "qnnpack_maxpool(): Expected kernel_size to be 2-dimensional: got ",
+         kernel_size.size());
+     TORCH_CHECK(
+         stride.size() == 2,
+         "qnnpack_maxpool(): Expected stride to be 2-dimensional: got ",
+         stride.size());
+     TORCH_CHECK(
+         dilation.size() == 2,
+         "qnnpack_maxpool(): Expected dilation to be 2-dimensional: got ",
+         dilation.size());
+     TORCH_CHECK(
+         padding.size() == 2,
+         "qnnpack_maxpool(): Expected padding to be 2-dimensional: got ",
+         padding.size());
+
+     int64_t batch_size = input.size(0);
+     int64_t inC = input.size(1);
+     int64_t inH = input.size(2);
+     int64_t inW = input.size(3);
+     // TODO: change it to contiguous(MemoryFormat::ChannelsLast) once a perf
+     // regression of it is fixed.
+     Tensor input_contig = input.permute({0, 2, 3, 1}).contiguous();
```

Comment:
@dzhulgakov please review this. I changed it to permute input tensor to NHWC similar to your changes for qconv.

Reply:
Yep, it should be fine